pytensor/tensor/slinalg.py

```diff
@@ -67,15 +68,27 @@ def make_node(self, x):
         return Apply(self, [x], [x.type()])
 
     def perform(self, node, inputs, outputs):
-        x = inputs[0]
-        z = outputs[0]
+        (x,) = inputs
+        (z,) = outputs
+        input_dtype = x.dtype
         try:
-            z[0] = scipy.linalg.cholesky(x, lower=self.lower).astype(x.dtype)
```

Comment:
Is it worth handling `F_CONTIGUOUS` too?  Or is `C_CONTIGUOUS` way more common?  Sorry if it's a dumb Q, I haven't fiddled with these flags in numpy before.  Like, if its not c_contiguous would it be a bad idea to force it to be here?

Reply:
`C_CONTIGUOUS` is the default for numpy. We have to set this flag because the actual cholesky routine is written in fortran, and it's really picky about the inputs. If they're not exactly right (including being in column-major order), then the inputs get copied before they're passed along to the routine and the in-place operation doesn't happen.

Reply:
That may explain a comment that I saw recently lying around some other scipy Op saying it didn't seem to be respecting the flag

Reply:
Does Pytensor make sure that all the arrays that get passed around always remain contiguous?

Reply:
No. There's an explicit Op for that. It's not guaranteed otherwise

Reply:
> No. There's an explicit Op for that. It's not guaranteed otherwise

An op that does the equivalent of `np.asfortranarray`?

Found the same thing with fortran / c-contiguous arrays for `sp.linalg.solve` btw.  `overwrite_a` only overwrites if `A` is f-contiguous.  Looks like this happens at the lapack call to getrf that does an LU decomposition, which must be fortran.  When `overwrite_b=True`, `b` is always overwritten whether `A` is f or c contiguous.

Reply:
> An op that does the equivalent of np.asfortranarray?

No, I meant we have an Op to make an array c-contiguous